lib/go-tc/users.go

```diff
@@ -86,10 +94,11 @@ type commonUserFields struct {
 
 // User fields in v14 have been updated to be nullable
 type User struct {
-	Username         *string    `json:"username" db:"username"`
-	RegistrationSent *TimeNoMod `json:"registrationSent" db:"registration_sent"`
-	LocalPassword    *string    `json:"localPasswd,omitempty" db:"local_passwd"`
-	RoleName         *string    `json:"roleName,omitempty" db:"-"`
+	Username             *string    `json:"username" db:"username"`
+	RegistrationSent     *TimeNoMod `json:"registrationSent" db:"registration_sent"`
+	LocalPassword        *string    `json:"localPasswd,omitempty" db:"local_passwd"`
+	ConfirmLocalPassword *string    `json:"confirmLocalPasswd,omitempty" db:"confirm_local_passwd"`
```

Comment:
So, Perl never checked that they matched but it *did* update it with a hash of the passed value when it was present and not an empty string. I've made updates so that it matches the Perl behavior by not checking that they match, not checking that both are present when either one is, and updating it whenever it's present and not empty. Is that acceptable?

Reply:
While I think we'd be okay completely ignoring it in the rewritten version, I think matching the behavior Perl version is acceptable as well.